lib/react_on_rails/engine.rb

```diff
@@ -5,8 +5,9 @@
 module ReactOnRails
   class Engine < ::Rails::Engine
     config.to_prepare do
-      if File.exist?(VersionChecker::NodePackageVersion.package_json_path)
-        VersionChecker.build.raise_if_gem_and_node_package_versions_differ
+      if VersionChecker.instance("package.json").raise_if_gem_and_node_package_versions_differ &&
+         VersionChecker.instance("client/package.json").raise_if_gem_and_node_package_versions_differ
+        Rails.logger.warn("No 'react-on-rails' entry found in 'dependencies' in package.json or client/package.json.")
```

Comment:
1. Isn't this a breaking change?
2. Semantically, having methods with side effect of `raise` does not seem right for an `if`

Reply:
all this consistency checking logic should go to one method, that logs warnings for now, and will throw for the next major version bump